graphql/schema.graphql

```diff
@@ -1,57 +1,95 @@
-"""User role"""
-enum Role {
-  """Administrator role (department staff etc.)"""
-  ADMIN
-  """User role (professor, student etc.)"""
-  USER
-}
-
 scalar Date
 
 type Query {
-  """Get the current user"""
+  """
+  Get the current user
+  """
   me: User
-  """Find a user by their id"""
+  """
+  Find a user by their id
+  """
   findUserById(id: ID!): User
-  """Get Teaching Preference Survey"""
+  """
+  Get Teaching Preference Survey
+  """
   survey: TeachingPreferenceSurvey!
-  """Get a list of courses for a given term"""
+  """
+  Get a list of courses for a given term
+  """
   courses(term: Term): [CourseSection!]
   """
   Schedule for a given term. If year is given, returns the most recent schedule generated for that year.
   """
   schedule(year: Int): Schedule
+  """
+  Get all courses preferences
+  """
+  coursePreferences: [CoursePreference!]
+  """
+  Get the all users
+  """
+  allUsers: [User!]
 }
 
-"""Generated schedule for a year"""
+"""
+Generated schedule for a year
+"""
 type Schedule {
-  """ID of the schedule"""
+  """
+  ID of the schedule
+  """
   id: ID!
-  """Year for the schedule"""
+  """
+  Year for the schedule
+  """
   year: Int!
-  """When the schedule was generated"""
+  """
+  When the schedule was generated
+  """
   createdAt: Date!
-  """Scheduled courses"""
+  """
+  Scheduled courses
+  """
   courses(term: Term!): [CourseSection!]
 }
 
 type Mutation {
-  """Login into a user account using email and password"""
+  """
+  Login into a user account using email and password
+  """
   login(username: String!, password: String!): AuthPayload!
-  """Logout the currently logged in user"""
+  """
+  Logout the currently logged in user
+  """
   logout: AuthPayload!
-  """Register a new user account"""
+  """
+  Register a new user account
+  """
   createUser(username: String!): CreateUserMutationResult!
-  """Updates a user given the user id."""
+  """
+  Updates a user given the user id.
+  """
   updateUser(input: UpdateUserInput!): UpdateUserMutationResult
-  """Change the password of the currently logged in user"""
+  """
+  Change the password of the currently logged in user
+  """
   changeUserPassword(input: ChangeUserPasswordInput!): Response!
-  """Reset a users password."""
+  """
+  Reset a users password.
+  """
   resetPassword(id: ID!): ResetPasswordMutationResult!
-  """Teaching preferences"""
+  """
+  Teaching preferences
+  """
   createTeachingPreference(input: CreateTeachingPreferenceInput!): Response!
-  """Generate schedule"""
+  """
+  Generate schedule
+  """
   generateSchedule(input: GenerateScheduleInput!): Response!
+  """
+  Edit schedule
+  """
+  editSchedule(year: Int!): Response!
```

Comment:
First of all, wouldn't you need to supply data to edit the schedule instead of just yeaR=r?

And if this is the solution to editable schedules on the frontend, I've been thinking maybe it makes more sense to edit the courses meeting times directly, instead of the whole schedule?
My understanding and correct me if I'm wrong is that a schedule object points to a bunch of different courses that make up the schedule. So if you edit the course that's being pointed to, it will still be reflected on the calendar but you don't need to edit the whole schedule.

Sorry for the long comment lmk if that makes sense

Reply:
Your point makes sense 👍

Reply:
TBH edit schedule is unrealistic to have working for the next demo, I think we should drop this entirely for now.

Reply:
Makes sense,  its more of editing the courses instead of editing the schedule. 

If we are editing courses, then it I think the input would be changing the course meeting times, section number, etc, so basically courseSection? instead of year

@Adamk8, would front end have any concerns? because I am not sure if editing the courses in a schedule would automatically update the calendar?